bundles/binding/org.openhab.binding.autelis/src/main/java/org/openhab/binding/autelis/internal/AutelisBinding.java

```diff
@@ -0,0 +1,353 @@
+
+	/** 
+	 * the refresh interval which is used to poll values from the Autelis
+	 * device (optional, defaults to 5000ms)
+	 */
+	private long refreshInterval = 5000;
+	
+	public AutelisBinding() {
+		logger.debug("Autelius binding started");
+	}
+		
+	/**
+	 * Called by the SCR to activate the component with its configuration read from CAS
+	 * 
+	 * @param bundleContext BundleContext of the Bundle that defines this component
+	 * @param configuration Configuration properties for this component obtained from the ConfigAdmin service
+	 */
+	public void activate(final BundleContext bundleContext, final Map<String, Object> configuration) {
+		configureBinding(configuration);
+	}
+	
+	/**
+	 * Called by the SCR when the configuration of a binding has been changed through the ConfigAdmin service.
+	 * @param configuration Updated configuration properties
+	 */
+	public void modified(final Map<String, Object> configuration) {
+		configureBinding(configuration);
+	}
+	
+	/**
+	 * Configures binding for both bundle activation and modification.
+	 * @param configuration
+	 */
+	private void configureBinding(final Map<String, Object> configuration){
+		String refreshIntervalString = (String) configuration.get("refresh");
+		if (StringUtils.isNotBlank(refreshIntervalString)) {
+			refreshInterval = Long.parseLong(refreshIntervalString);
+		}
+		
+		int port = DEFAULT_PORT;
+		String host = (String) configuration.get("host");
+		String username = (String) configuration.get("username");
+		String password = (String) configuration.get("password");
+		
+		if(StringUtils.isBlank(host)){
+			logger.error("Host config parameter is missing");
+			setProperlyConfigured(false);
+			return;
+		}
+		
+		String portString = (String) configuration.get("port"); 
+		if(StringUtils.isNotBlank(portString))
+			port = Integer.parseInt(portString);
+
+		String prefix = "";
+		if(username != null) {
+			prefix = username + ":" + password + "@";
+		}
+		
+		baseURL = "http://" + prefix + host + ":" + port;
+
+		logger.debug("Autelius binding configured for host {}", host);
+		
+		setProperlyConfigured(true);
+	}
+	
+	/**
+	 * @{inheritDoc}
+	 */
+	@Override
+	protected long getRefreshInterval() {
+		return refreshInterval;
+	}
+
+	/**
+	 * @{inheritDoc}
+	 */
+	@Override
+	protected String getName() {
+		return "autelis";
+	}
+	
+	/**
+	 * @{inheritDoc}
+	 */
+	@Override
+	protected void execute() {
+		logger.trace("Connecting to {}" + baseURL);	
+
+		clearState();
+		
+		//we will reconstruct the document with all the responses combined for XPATH
+		StringBuilder sb = new StringBuilder("<response>");
+		
+		//pull down the three xml documents
+		String [] statuses = {"status", "chem", "pumps"};
+		for(String status : statuses){
+			String response = HttpUtil.executeUrl("GET", baseURL + "/" + status + ".xml", TIMEOUT);
+			logger.trace(baseURL + "/" + status + ".xml \n {}", response);
+			if(response == null){
+				logger.error("No response from Autelis controller!");
```

Comment:
decrease to WARN since the overall openHAB stability is not affected by this issue